.travis.yml

```diff
@@ -1,14 +1,25 @@
 language: elixir
-matrix:
+
+sudo: false
+
+otp_release: '21.1'
+elixir: '1.8'
+
+stages:
+  - check formatted
+  - certdata
+  - test
+
+jobs:
   include:
-  - elixir: 1.7.4
-    otp_release: 21.1
-  - elixir: 1.0.5
-    otp_release: 17.5
-script:
-  - if [[ `elixir -v` = *"1.7"* ]]; then mix format --check-formatted; fi
-  - if [[ `elixir -v` = *"1.7"* ]]; then mix certdata; fi
-  - mix test
-notifications:
-  recipients:
-    - eric.meadows.jonsson@gmail.com
+    - stage: test
+      script: mix test --trace
+
+    - otp_release: '17.5'
```

Comment:
Should this be on the same level as `- stage`?

Reply:
For which stages does it apply?

Reply:
Yes. I know it's hacky, but the way this works is that each item in the array at this level is a "stage" if it contains the `stage` key, otherwise it's the previous stage with some things changes. So basically by having this we're saying "inherit everything from the `stage: test` stage, but change the `otp_release` and `elixir` keys. Does that make sense?

Reply:
Btw you can see that in practice this works in Travis, where we only have one "test" stage with multiple steps for the different versions.

Reply:
It makes sense but it's a bit confusing :)

Reply:
It's terrible. I couldn't find a better way of doing this because afaik they still don't support a stage with a matrix (which is what we want here). 😕